src/RepoToolset/Tools.props

```diff
@@ -1,9 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
 <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
 <Project>
-  <ItemGroup>
-    <PackageReference Include="Microsoft.Net.Compilers" Version="$(MicrosoftNetCompilersVersion)" />
-  </ItemGroup>
+
   <PropertyGroup>
-    <ToolsetCompilerPackageDir>$(NuGetPackageRoot)microsoft.net.compilers\$(MicrosoftNetCompilersVersion)\</ToolsetCompilerPackageDir>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
   </PropertyGroup>
+
+  <Choose>
+    <When Condition="'$(UsingToolMicrosoftNetCompilers)' != 'netcore'">
```

Comment:
The values for this are true, false, netcore? That's confusing.

Reply:
+1

When do we need to use netcore compilers? When building on Linux/MacOS? We can detect that condition automatically, instead of requiring the repo to chose manually.

---
In reply to: [145545980](https://github.com/dotnet/roslyn-tools/pull/60#discussion_r145545980) [](ancestors = 145545980)

Reply:
Right now if we run core msbuild, we can't build satellites with 2.0 SDK. (Fixed in 2.1+, but source-build won't be using that just yet.). So the condition would have to be MSBuildRuntimeType == Core, but then it would slow down builds that can use dotnet + desktop csc because core csc is quite a bit slower.

Reply:
Do you have a better suggestion. You don't want to allow importing both, and the only other option that I can think of is a task to validate both `UsingToolMicrosoftNetCompilers` and `UsingToolMicrosoftNetCompilersNetCore` are not simultaneously true

Reply:
Yes, see my comment below. Add ```UsingToolNetCoreCompilers``` property. If true use the compiler that comes with dotnet SDK, otherwise use Microsoft.Net.Compilers package.

Reply:
What about when building with desktop msbuild and UsingToolNetCoreCompilers=true. Is that an error case?

Reply:
I'd think so.